automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/azure/AzureDiskSpec.scala

```diff
@@ -0,0 +1,189 @@
+package org.broadinstitute.dsde.workbench.leonardo.azure
+
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.broadinstitute.dsde.workbench.google2.streamUntilDoneOrTimeout
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import org.broadinstitute.dsde.workbench.GeneratedLeonardoClient
+import org.broadinstitute.dsde.workbench.client.leonardo.model.{
+  AzureDiskConfig,
+  ClusterStatus,
+  CreateAzureRuntimeRequest,
+  DiskStatus,
+  GetRuntimeResponse
+}
+import org.broadinstitute.dsde.workbench.leonardo.LeonardoTestTags.ExcludeFromJenkins
+import org.broadinstitute.dsde.workbench.leonardo.TestUser.Hermione
+import org.scalatest.{ParallelTestExecution, Retries}
+import org.broadinstitute.dsde.workbench.service.test.CleanUp
+import org.broadinstitute.dsde.workbench.leonardo.{AzureBillingBeforeAndAfter, LeonardoTestUtils}
+
+import scala.concurrent.duration._
+
+class AzureDiskSpec
+    extends AzureBillingBeforeAndAfter
+    with LeonardoTestUtils
+    with ParallelTestExecution
+    with TableDrivenPropertyChecks
+    with Retries
+    with CleanUp {
+
+  implicit val accessToken = Hermione.authToken()
+
+  "create a disk, keep it on runtime delete, and then attach it to a new runtime" taggedAs ExcludeFromJenkins in {
+    workspaceDetails =>
+      val workspaceId = workspaceDetails.workspace.workspaceId
+
+      val labelMap: java.util.HashMap[String, String] = new java.util.HashMap[String, String]()
+      labelMap.put("automation", "true")
+
+      val runtimeName = randomClusterName
+      val runtimeName2 = randomClusterName
```

Comment:
It does haha, syntax is a bit odd but this is indeed how you call an argument-less function. Because it is defined as `def randomClusterName` not `val randomClusterName` the right hand is recomputed each time.